tests/epoch.rs

```diff
@@ -1337,3 +1343,207 @@ fn test_minmax() {
+    // 2022/12/01 was a thursday
+    let epoch = Epoch::from_gregorian_utc_at_midnight(2022, 12, 01);
+    assert_eq!(epoch.weekday_utc(), Weekday::Thursday);
+    permutate_time_scale(epoch, Weekday::Thursday);
+    // 2022/11/28 was a monday
+    let epoch = Epoch::from_gregorian_utc_at_midnight(2022, 11, 28);
+    assert_eq!(epoch.weekday_utc(), Weekday::Monday);
+    permutate_time_scale(epoch, Weekday::Monday);
+    // 1988/01/02 was a Saturday
+    let epoch = Epoch::from_gregorian_utc_at_midnight(1988, 1, 2);
+    assert_eq!(epoch.weekday_utc(), Weekday::Saturday);
+    permutate_time_scale(epoch, Weekday::Saturday);
+}
+
+#[test]
+fn test_get_time() {
+    let epoch = Epoch::from_gregorian_utc(2022, 12, 01, 10, 11, 12, 13);
+    assert_eq!(epoch.hours(), 10);
+    assert_eq!(epoch.minutes(), 11);
+    assert_eq!(epoch.seconds(), 12);
+    assert_eq!(epoch.milliseconds(), 0);
+    assert_eq!(epoch.microseconds(), 0);
+    assert_eq!(epoch.nanoseconds(), 13);
+
+    let epoch_midnight = epoch.with_hms(0, 0, 0);
+    assert_eq!(
+        epoch_midnight,
+        Epoch::from_gregorian_utc_at_midnight(2022, 12, 01) + 13 * Unit::Nanosecond
+    );
+
+    let epoch_midnight = epoch.with_hms_strict(0, 0, 0);
+    assert_eq!(
+        epoch_midnight,
+        Epoch::from_gregorian_utc_at_midnight(2022, 12, 01)
+    );
+}
+
+#[test]
+fn test_start_of_week() {
+    // 2022/12/01 + some offset, was a thursday
+    let epoch = Epoch::from_gregorian_utc(2022, 12, 01, 10, 11, 12, 13);
+    assert_eq!(epoch.weekday_utc(), Weekday::Thursday);
+    // 2022/11/27 was the related sunday / start of week
+    assert_eq!(
+        epoch.previous_weekday_at_midnight(Weekday::Sunday),
+        Epoch::from_gregorian_utc_at_midnight(2022, 11, 27)
+    );
+    assert_eq!(
+        epoch
+            .previous_weekday_at_midnight(Weekday::Sunday)
+            .weekday_utc(),
+        Weekday::Sunday
+    );
+
+    let epoch = Epoch::from_gregorian_utc(2022, 09, 15, 01, 01, 01, 01);
+    assert_eq!(epoch.weekday_utc(), Weekday::Thursday);
+    assert_eq!(
+        epoch.previous_weekday_at_midnight(Weekday::Sunday),
+        Epoch::from_gregorian_utc_at_midnight(2022, 09, 11)
+    );
+    assert_eq!(
+        epoch
+            .previous_weekday_at_midnight(Weekday::Sunday)
+            .weekday_utc(),
+        Weekday::Sunday
+    );
+}
+
+#[test]
+fn test_time_of_week() {
+    // GPST
+    // https://www.labsat.co.uk/index.php/en/gps-time-calculator
+    // 01/12/2022 00:00:00 <=> (2238, 345_618_000_000_000)
+    //      2238 weeks since 1980 + 345_600_000_000_000 ns since previous Sunday
+    //                            +      18_000_000_000 ns for elapsed leap seconds
+    let epoch = Epoch::from_time_of_week(2238, 345_618_000_000_000, TimeScale::GPST);
+    assert_eq!(epoch.to_gregorian_utc(), (2022, 12, 01, 00, 00, 00, 00));
+    assert_eq!(epoch.to_time_of_week(), (2238, 345_618_000_000_000));
+
+    let epoch_utc = epoch.in_time_scale(TimeScale::UTC);
```

Comment:
Hello Chris,
I just pushed more tests, especially using other TimeScales it works great.

But this test right here: switching to UTC and asking for a reciprocal, always ends up in a perfect +1day offset. %H:%M:%S:%ns fields are correct. 
Starting from a different timescale than GPST, GST or BDT for example, produce the same effect.

Reply:
Okay, I'll investigate. This might be a sign of an error when changing time scales, I'm not sure.